banking-system(without-DB)/class/BankAccount.php

```diff
@@ -0,0 +1,38 @@
+<?php
+require_once 'Logger.php';
+
+abstract class BankAccount
+{
+    use Logger;
+    protected int $accountNumber;
+    protected float $balance = 0;
+    protected static int $totalAccounts = 0;
+
+    public function __construct(int $accountNumber, float $initialBalance = 0)
+    {
+        $this->accountNumber = $accountNumber;
+        $this->balance = $initialBalance;
+        self::$totalAccounts++;
+        echo $this->log("---------------new account created-------------------");
```

Comment:
```suggestion
         $this->log("---------------new account created-------------------");
```
log function already has echo.